[bitnami/grafana-operator] `ValidationError(Grafana.spec.dashboardNamespaceSelector`

### Name and Version

bitnami/grafana-operator 2.2.9/2.2.10

### What steps will reproduce the bug?

With this simple values.yaml file:
```yaml
grafana:
  dashboardNamespaceSelector:
    - matchLabels:
        app.kubernetes.io/name: monitoring
```
And these env variables:
```
❯ cat chart.sh
#!/bin/bash

CHART_REPO_URL="https://charts.bitnami.com/bitnami"
CHART_REPO_NAME="bitnami"

CHART_NAMESPACE="monitoring"
CHART_NAME="grafana-operator"
CHART_VERSION="2.2.10"
❯ . ./chart.sh
```

I get the following error:
```
❯ helm upgrade --install $CHART_NAME -n $CHART_NAMESPACE -f values/exploitation/values.yaml $CHART_REPO_NAME/$CHART_NAME --version $CHART_VERSION
Release "grafana-operator" does not exist. Installing it now.
Error: unable to build kubernetes objects from release manifest: error validating "": error validating data: ValidationError(Grafana.spec.dashboardNamespaceSelector): invalid type for org.integreatly.v1alpha1.Grafana.spec.dashboardNamespaceSelector: got "array", expected "map"
```

### Are you using any custom parameters or values?

_No response_

### What is the expected behavior?

_No response_

### What do you see instead?

Debug trace:
```
❯ helm upgrade --install $CHART_NAME -n $CHART_NAMESPACE -f values/exploitation/values.yaml $CHART_REPO_NAME/$CHART_NAME --version $CHART_VERSION --debug
history.go:56: [debug] getting history for release grafana-operator
Release "grafana-operator" does not exist. Installing it now.
install.go:178: [debug] Original chart version: "2.2.10"
install.go:199: [debug] CHART PATH: /home/baptiste/.cache/helm/repository/grafana-operator-2.2.10.tgz

client.go:128: [debug] creating 1 resource(s)
install.go:151: [debug] CRD grafanadashboards.integreatly.org is already present. Skipping.
client.go:128: [debug] creating 1 resource(s)
install.go:151: [debug] CRD grafanadatasources.integreatly.org is already present. Skipping.
client.go:128: [debug] creating 1 resource(s)
install.go:151: [debug] CRD grafananotificationchannels.integreatly.org is already present. Skipping.
client.go:128: [debug] creating 1 resource(s)
install.go:151: [debug] CRD grafanas.integreatly.org is already present. Skipping.
Error: unable to build kubernetes objects from release manifest: error validating "": error validating data: ValidationError(Grafana.spec.dashboardNamespaceSelector): invalid type for org.integreatly.v1alpha1.Grafana.spec.dashboardNamespaceSelector: got "array", expected "map"
helm.go:88: [debug] error validating "": error validating data: ValidationError(Grafana.spec.dashboardNamespaceSelector): invalid type for org.integreatly.v1alpha1.Grafana.spec.dashboardNamespaceSelector: got "array", expected "map"
helm.sh/helm/v3/pkg/kube.scrubValidationError
        helm.sh/helm/v3/pkg/kube/client.go:631
helm.sh/helm/v3/pkg/kube.(*Client).Build
        helm.sh/helm/v3/pkg/kube/client.go:204
helm.sh/helm/v3/pkg/action.(*Install).RunWithContext
        helm.sh/helm/v3/pkg/action/install.go:275
main.runInstall
        helm.sh/helm/v3/cmd/helm/install.go:265
main.newUpgradeCmd.func2
        helm.sh/helm/v3/cmd/helm/upgrade.go:124
github.com/spf13/cobra.(*Command).execute
        github.com/spf13/cobra@v1.2.1/command.go:856
github.com/spf13/cobra.(*Command).ExecuteC
        github.com/spf13/cobra@v1.2.1/command.go:974
github.com/spf13/cobra.(*Command).Execute
        github.com/spf13/cobra@v1.2.1/command.go:902
main.main
        helm.sh/helm/v3/cmd/helm/helm.go:87
runtime.main
        runtime/proc.go:225
runtime.goexit
        runtime/asm_amd64.s:1371
unable to build kubernetes objects from release manifest
helm.sh/helm/v3/pkg/action.(*Install).RunWithContext
        helm.sh/helm/v3/pkg/action/install.go:277
main.runInstall
        helm.sh/helm/v3/cmd/helm/install.go:265
main.newUpgradeCmd.func2
        helm.sh/helm/v3/cmd/helm/upgrade.go:124
github.com/spf13/cobra.(*Command).execute
        github.com/spf13/cobra@v1.2.1/command.go:856
github.com/spf13/cobra.(*Command).ExecuteC
        github.com/spf13/cobra@v1.2.1/command.go:974
github.com/spf13/cobra.(*Command).Execute
        github.com/spf13/cobra@v1.2.1/command.go:902
main.main
        helm.sh/helm/v3/cmd/helm/helm.go:87
runtime.main
        runtime/proc.go:225
runtime.goexit
        runtime/asm_amd64.s:1371
```

### Additional information

_No response_

Hi @javsalgar,
Thanks for your suggestion, you're right the grafana operator is expecting a dict: https://github.com/grafana-operator/grafana-operator/blob/eec0e7c38193526faa8011816394d748a6727cca/controllers/common/controllerState.go

I got these warnings but it went through:
```
❯ helm upgrade --install --dry-run $CHART_NAME -n $CHART_NAMESPACE -f manifests/monitoring/grafana-operator/values/prod/values.yaml $CHART_REPO_NAME/$CHART_NAME --version $CHART_VERSION
Release "grafana-operator" does not exist. Installing it now.
coalesce.go:202: warning: destination for dashboardNamespaceSelector is a table. Ignoring non-table value []
coalesce.go:202: warning: destination for dashboardNamespaceSelector is a table. Ignoring non-table value []
```
```
❯ k -n monitoring get grafana grafana-operator-grafana -oyaml | yq -e '.spec.dashboardNamespaceSelector'
matchLabels:
  app.kubernetes.io/name: monitoring
```
I don't know where those warnings come from and if we can get rid of these.

The values.yaml example should be corrected with your suggestion: https://github.com/bitnami/charts/blob/master/bitnami/grafana-operator/values.yaml#L680

--- a/bitnami/grafana-operator/Chart.yaml
+++ b/bitnami/grafana-operator/Chart.yaml
@@ -24,4 +24,4 @@ name: grafana-operator
 sources:
   - https://github.com/grafana-operator/grafana-operator
   - https://github.com/bitnami/bitnami-docker-grafana-operator
-version: 2.2.11
+version: 2.2.12

--- a/bitnami/grafana-operator/README.md
+++ b/bitnami/grafana-operator/README.md
@@ -234,8 +234,8 @@ For more information, refer to the [documentation on the differences between the
 | `grafana.configMaps`                                        | Extra configMaps to mount into the grafana pod                                                | `[]`                     |
 | `grafana.secrets`                                           | Extra secrets to mount into the grafana pod                                                   | `[]`                     |
 | `grafana.jsonnetLibrarySelector`                            | Configuring the read for jsonnetLibraries to pull in.                                         | `{}`                     |
-| `grafana.dashboardLabelSelectors`                           | This selects dashboards on the label.                                                         | `{}`                     |
-| `grafana.dashboardNamespaceSelector`                        | Watch for dashboards only in the Namespaces that have the specified namespace label           | `[]`                     |
+| `grafana.dashboardLabelSelectors`                           | This selects dashboards on the label.                                                         | `[]`                     |
+| `grafana.dashboardNamespaceSelector`                        | Watch for dashboards only in the Namespaces that have the specified namespace label           | `{}`                     |
 | `grafana.livenessProbe.enabled`                             | Enable livenessProbe                                                                          | `true`                   |
 | `grafana.livenessProbe.initialDelaySeconds`                 | Initial delay seconds for livenessProbe                                                       | `120`                    |
 | `grafana.livenessProbe.periodSeconds`                       | Period seconds for livenessProbe                                                              | `10`                     |

--- a/bitnami/grafana-operator/values.yaml
+++ b/bitnami/grafana-operator/values.yaml
@@ -677,10 +677,10 @@ grafana:
   ## Ref: https://github.com/grafana-operator/grafana-operator/blob/master/documentation/multi_namespace_support.md#2-dashboardnamespaceselector
   ## e.g:
   ## dashboardNamespaceSelector:
-  ##   - matchLabels:
+  ##   matchLabels:
   ##     key: value
   ##
-  dashboardNamespaceSelector: []
+  dashboardNamespaceSelector: {}
   ## Grafana containers' liveness probe
   ## Ref: https://kubernetes.io/docs/concepts/workloads/pods/pod-lifecycle/#container-probes
   ## @param grafana.livenessProbe.enabled Enable livenessProbe

